src/constructs/cloudwatch/lambda-alarms.ts

```diff
@@ -46,3 +46,49 @@ export class GuLambdaErrorPercentageAlarm extends GuAlarm {
     super(scope, id, alarmProps);
   }
 }
+
+export interface GuLambdaThrottlingMonitoringProps
+  extends Omit<
+    GuAlarmProps,
+    "metric" | "threshold" | "comparisonOperator" | "evaluationPeriods" | "treatMissingData" | "app"
+  > {
+  /**
+   * Sum of thottled invocations at which to alarm.
+   *
+   * @defaultValue 1
+   */
+  toleratedThrottlingCount?: number;
+
+  /**
+   * Evaluation period in minutes for alarm.
+   *
+   * @defaultValue 1
+   */
+  numberOfMinutesAboveThresholdBeforeAlarm?: number;
+
+  noMonitoring?: false;
+}
+
+interface GuLambdaThrottlingAlarmProps extends GuLambdaThrottlingMonitoringProps {
+  lambda: GuLambdaFunction;
+}
+
+export class GuLambdaThrottlingAlarm extends GuAlarm {
+  constructor(scope: GuStack, id: string, props: GuLambdaThrottlingAlarmProps) {
+    super(scope, id, {
+      ...props,
+      app: props.lambda.app,
+      alarmName: props.alarmName ?? "Lambda throttling alarm",
```

Comment:
I think we might want to put the stage into the default alarm name (I'm 99% sure alarm names must be unique within the account/region i.e. I don't think you can deploy an alarm with the same name via `CODE` and `PROD` stacks).